src/Module/Magic.php

```diff
@@ -71,41 +71,76 @@ protected function rawContent(array $request = [])
 			$this->logger->info('No contact record found', $_REQUEST);
-			// @TODO Finding a more elegant possibility to redirect to either internal or external URL
-			$this->app->redirect($dest);
-		}
-		$contact = $this->dba->selectFirst('contact', ['id', 'nurl', 'url'], ['id' => $cid]);
 
-		// Redirect if the contact is already authenticated on this site.
-		if ($this->app->getContactId() && strpos($contact['nurl'], Strings::normaliseLink($this->baseUrl)) !== false) {
-			$this->logger->info('Contact is already authenticated');
-			System::externalRedirect($dest);
+			if (!$owa) {
+				// @TODO Finding a more elegant possibility to redirect to either internal or external URL
+				$this->app->redirect($dest);
+			}
+		} else {
+			$contact = $this->dba->selectFirst('contact', ['id', 'nurl', 'url'], ['id' => $cid]);
+
+			// Redirect if the contact is already authenticated on this site.
+			if ($this->app->getContactId() && strpos($contact['nurl'], Strings::normaliseLink($this->baseUrl)) !== false) {
+				$this->logger->info('Contact is already authenticated');
+				System::externalRedirect($dest);
+			}
+
+			$this->logger->info('Contact URL: ', ['url' => $contact['url']]);
 		}
 
 		// OpenWebAuth
 		if ($this->userSession->getLocalUserId() && $owa) {
+			$this->logger->info('Checking OWA now');
 			$user = User::getById($this->userSession->getLocalUserId());
 
-			// Extract the basepath
-			// NOTE: we need another solution because this does only work
-			// for friendica contacts :-/ . We should have the basepath
-			// of a contact also in the contact table.
-			$exp = explode('/profile/', $contact['url']);
-			$basepath = $exp[0];
+			$basepath = false;
+			if (!empty($contact)) {
+				$this->logger->info('Contact found - trying friendica style basepath extraction');
+				// Extract the basepath
+				// NOTE: we need another solution because this does only work
+				// for friendica contacts :-/ . We should have the basepath
+				// of a contact also in the contact table.
+				$contact_url = $contact['url'];
+				if (!(strpos($contact_url, '/profile/') === false)) {
+					$exp = explode('/profile/', $contact['url']);
+					$basepath = $exp[0];
+					$this->logger->info('Basepath: ', ['basepath' => $basepath]);
+				} else {
+					$this->logger->info('Not possible to extract basepath in friendica style');
+				}
+			}
+			if (!$basepath) {
+				// For the rest of the OpenWebAuth-enabled Fediverse
+				$parsed = parse_url($dest);
+				$this->logger->info('Parsed URL: ', ['parsed URL' => $parsed]);
+				if (!$parsed) {
+					System::externalRedirect($dest);
+				}
+				$basepath = $parsed['scheme'] . '://' . $parsed['host'] . (isset($parsed['port']) ? ':' . $parsed['port'] : '');
+			}
 
+			$accept_headers = ['application/x-dfrn+json', 'application/x-zot+json'];
 			$header = [
-				'Accept'		  => ['application/x-dfrn+json', 'application/x-zot+json'],
+				'Accept'		  => $accept_headers, // ['application/x-dfrn+json', 'application/x-zot+json'],
```

Comment:
Please remove the redundant comment.
```suggestion
				'Accept'         => $accept_headers,
```